src/D2L.Security.OAuth2/Keys/Default/TokenSigner.cs

```diff
@@ -1,39 +1,49 @@
-﻿using System.IdentityModel.Tokens;
+﻿using System.Collections.Generic;
+using System.IdentityModel.Tokens;
+using System.Security.Claims;
 using System.Threading.Tasks;
 using D2L.Services;
 
 namespace D2L.Security.OAuth2.Keys.Default {
-	internal sealed class TokenSigner : ITokenSigner {
-
-		private readonly IPrivateKeyProvider m_privateKeyProvider;
-
-		public TokenSigner(
-			IPrivateKeyProvider privateKeyProvider
-		) {
-			m_privateKeyProvider = privateKeyProvider;
-		}
-
-		async Task<string> ITokenSigner.SignAsync( UnsignedToken token ) {
-			JwtSecurityToken jwt;
-			using( D2LSecurityToken securityToken = await m_privateKeyProvider
-				.GetSigningCredentialsAsync()
-				.SafeAsync()
-			) {
-				jwt = new JwtSecurityToken(
-					issuer: token.Issuer,
-					audience: token.Audience,
-					claims: token.Claims,
-					notBefore: token.NotBefore,
-					expires: token.ExpiresAt,
-					signingCredentials: securityToken.GetSigningCredentials()
-				);
-
-				var jwtHandler = new JwtSecurityTokenHandler();
-
-				string signedRawToken = jwtHandler.WriteToken( jwt );
-
-				return signedRawToken;
-			}
-		}
-	}
+    internal sealed class TokenSigner : ITokenSigner {
+
+        private readonly IPrivateKeyProvider m_privateKeyProvider;
+
+        public TokenSigner(
+            IPrivateKeyProvider privateKeyProvider
+        ) {
+            m_privateKeyProvider = privateKeyProvider;
+        }
+
+        async Task<string> ITokenSigner.SignAsync( UnsignedToken token ) {
+            JwtSecurityToken jwt;
+            using( D2LSecurityToken securityToken = await m_privateKeyProvider
+                .GetSigningCredentialsAsync()
+                .SafeAsync()
+            ) {
+                jwt = new JwtSecurityToken(
+                    issuer: token.Issuer,
+                    audience: token.Audience,
+                    claims: new List<Claim>(),
+                    notBefore: token.NotBefore,
+                    expires: token.ExpiresAt,
+                    signingCredentials: securityToken.GetSigningCredentials()
+                );
+
+                var claims = token.Claims;
+                foreach( var claim in claims ) {
+                    if( jwt.Payload.ContainsKey( claim.Key ) ) {

```

Comment:
This can't really occur _within_ the dictionary, so this just becomes "built-in" claims like `aud`, `iss`, `exp` etc. I think this should probably throw. What would happen before?

Reply:
It was throwing at this point in some of the unit tests, hence the dupe check.